cronjobs/add_authority_external_ref._automation.py

```diff
@@ -0,0 +1,183 @@
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Process logic:
+1. Check whether the gnd file "authorities-gnd-person_lds_*.jsonld.gz" on https://data.dnb.de/opendata/ is newer
+    than the last successful parse to gnd_wiki.csv. 
+    To doing this, there is a config file "/mnt/ZE020150/FID-Entwicklung/ub_tools/config_file_add_authority_ext_ref.cnf" that contain 2 lines of information the first line for the lastest successful date generating gnd_wiki data and the second line is the information about the lastest version on the web.
+    If the date of successful generating gnd_wiki (on first line) is older compare with the one on the web then:
+    a. Download the newer file from the web and put it into folder "/tmp"
+    b. The download file is a zip file, it needs to be extracted first
+    c. Get the information needed by gnd_wiki and put it into the file "/tmp/input_file_for_add_authority_external_ref.txt" using jq:
+        jq -c --stream '.' < authorities-gnd-person_lds.jsonld |grep -E 'https\:/\/d-nb\.info\/gnd\/|wikidata|wikipedia' > input_file_for_add_authority_external_ref.txt
+    d. Run "add_authority_external_ref" program. This program will check whether the lastest version date is newer compare to the lastest successful date generating gnd_wiki, if so then generate a new gnd_wiki.csv file. If it success generating gnd_wiki file then update the date on the config file.
+    e. Update the information in the config file
+
+"""
+
+import os
+import re
+import requests
+import util
+from urllib.request import urlopen
+import traceback
+from tqdm import tqdm
+import functools
+import shutil
+
+# this script needs:
+# 1. requests
+# 2. tqdm
+
+# Start global variable
+
+url = "https://data.dnb.de/opendata/"
+config_file = "/mnt/ZE020150/FID-Entwicklung/ub_tools/config_file_add_authority_ext_ref.cnf"
+share_folder = "/mnt/ZE020150/FID-Entwicklung/ub_tools/"
+# config_file = "/tmp/config_file_add_authority_ext_ref.cnf"
+# share_folder = "/tmp/"
+newer_file_name = ""
+gnd_wiki_file = "gnd_to_wiki.csv"
+current_file_date_integer = 0
+last_file_update_date_int = 0
+input_file_name_for_add_auth = "input_file_for_add_authority_external_ref.txt"
+
+
+# End global variable
+
+
+def DownloadTheFile(file_name):
+    url_path = f'{url}{file_name}'
+    target_file = f'/tmp/{file_name}'
+    dw = requests.get(url_path, stream=True)
```

Comment:
Would it be better to re-use one of the functions from util.py for downloading a single file? E.g. `WgetFetch` or `RetrieveFileByUrl`

Reply:
in these two functions, I don't know how to determine the output file (downloaded file)

Reply:
I think they will store the file in the current working directory, but i guess it would be better if we had a function that just takes two parameters e.g. the URL and the full local target file path, and downloads that. A simple way to do tat would be e.g. to add a second optional parameter to WgetFetch which also passed the target path as additional `-O ` parameter if the target path is not `None`. What do you think?